nomad/structs/structs.go

```diff
@@ -4744,10 +4776,42 @@ func (tg *TaskGroup) Validate(j *Job) error {
 		}
 	}
 
-	// Check for duplicate tasks, that there is only leader task if any,
-	// and no duplicated static ports
-	tasks := make(map[string]int)
+	portLabels := make(map[string]string)
 	staticPorts := make(map[int]string)
+	mappedPorts := make(map[int]string)
+
+	for _, net := range tg.Networks {
```

Comment:
This validator is getting pretty complicated - It might be worth breaking the network validation out into smaller fns just to keep this skimmable.

Reply:
Yeah theres some state that has to be kept between iterating through the tg networks and each task's networks that makes pulling this apart hard. I might split it up and just loop through the tasks twice (once for network validation in a separate func and another for validating the task).